### PR TITLE
Documentation - Variable types

### PR DESCRIPTION
**Pull request by mbabker**
To better help developers, I've started reviewing the "undefined methods" inspection on PhpStorm and trying to update docblocks to more accurately reflect what each parameter type is and what is being returned.  Most of these changes are based on what is commented in the docblock or based on the instance type that is being returned.

### CODE REVIEW

File: libraries/joomla/html/parameter/element/calendar.php [hunk 1]
```diff
@@ -31,10 +31,10 @@ class JElementCalendar extends JElement
 	/**
 	 * Fetch a calendar element
 	 *
-	 * @param   string  $name          Field name
-	 * @param   string  $value         The date value
-	 * @param   object  &$node         JSimpleXMLElement node object containing the settings for the element
-	 * @param   string  $control_name  Control name
+	 * @param   string             $name          Field name
+	 * @param   string             $value         The date value
+	 * @param   JSimpleXMLElement  &$node         JSimpleXMLElement node object containing the settings for the element
```
Comment (elkuku) [inline libraries/joomla/html/parameter/element/calendar.php]: I believe this should be JXMLElement instead of JSimpleXMLElement

Comment (mbabker) [inline libraries/joomla/html/parameter/element/calendar.php]: Possibly.  I'd have to get an environment set up where I'm calling JElement instances to accurately figure it out, but I'm hoping that in this instance (and the other few instances that are "properly" commented), the comment stating it's a JSimpleXMLElement is correct.

Comment (elkuku) [inline libraries/joomla/html/parameter/element/calendar.php]: I believe the others are also incorrect - I just don't wanted to spam ;)
JSimpleXMLElement has been deprecated and the functionality is now covered by JXMLElement which extends SimpleXMLElement (the PHP one).

**Comment by eddieajau** [comment]
Thanks Michael.  That's awesome work.

**Pull request merged**